### PR TITLE
osv-scanner 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by bashofmann**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/osv-scanner.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://github.com/google/osv-scanner.git",
+      tag:      "v1.0.0",
+      revision: "c6ed0fabdbe8ad75465bd7eab81ba904a9c89f42"
+  license "Apache-2.0"
+  head "https://github.com/google/osv-scanner.git", branch: "main"
+
+  depends_on "go" => [:build]
+
+  def install
+    system "go", "build", *std_go_args, "-o", bin/"osv-scanner", "./cmd/osv-scanner/"
```
Comment (SMillerDev) [inline Formula/osv-scanner.rb:13]:
```suggestion
    system "go", "build", *std_go_args, "./cmd/osv-scanner/"
```
This is a standard argument

File: Formula/osv-scanner.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class OsvScanner < Formula
+  desc "Vulnerability scanner which uses the OSV database"
+  homepage "https://github.com/google/osv-scanner"
+  url "https://github.com/google/osv-scanner.git",
+      tag:      "v1.0.0",
+      revision: "c6ed0fabdbe8ad75465bd7eab81ba904a9c89f42"
+  license "Apache-2.0"
+  head "https://github.com/google/osv-scanner.git", branch: "main"
+
+  depends_on "go" => [:build]
```
Comment (chenrui333) [inline Formula/osv-scanner.rb:10]:
```suggestion
  depends_on "go" => :build
```

File: Formula/osv-scanner.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://github.com/google/osv-scanner.git",
+      tag:      "v1.0.0",
+      revision: "c6ed0fabdbe8ad75465bd7eab81ba904a9c89f42"
+  license "Apache-2.0"
+  head "https://github.com/google/osv-scanner.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/osv-scanner/"
```
Comment (chenrui333) [inline Formula/osv-scanner.rb:13]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/osv-scanner"
```

File: Formula/osv-scanner.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    (testpath/"go.mod").write <<~EOS
+      module my-library
+
+      require (
+        github.com/BurntSushi/toml v1.0.0
+      )
+    EOS
+
+    scan_output = shell_output("#{bin}/osv-scanner --lockfile #{testpath}/go.mod")
+    assert_match "found 1 packages", scan_output
```
Comment (chenrui333) [inline Formula/osv-scanner.rb:26]:
```suggestion
    scan_output = shell_output("#{bin}/osv-scanner --lockfile #{testpath}/go.mod").strip
    assert_equal "Scanned #{testpath}/go.mod file and found 1 packages", scan_output
```

File: Formula/osv-scanner.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+class OsvScanner < Formula
+  desc "Vulnerability scanner which uses the OSV database"
+  homepage "https://github.com/google/osv-scanner"
+  url "https://github.com/google/osv-scanner.git",
+      tag:      "v1.0.0",
+      revision: "c6ed0fabdbe8ad75465bd7eab81ba904a9c89f42"
```
Comment (chenrui333) [inline Formula/osv-scanner.rb:6]:
```suggestion
  url "https://github.com/google/osv-scanner/archive/refs/tags/v1.0.0.tar.gz"
  sha256 "bc0f3a0736c5a6b1ed4461541dd0c0bb6b64e03fd702a97f267c595eff5a3f23"
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@bashofmann, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**